src/XMakeBuildEngine/Definition/Project.cs

```diff
@@ -1032,6 +1033,170 @@ public static string GetEvaluatedItemIncludeEscaped(ProjectItemDefinition item)
             return ((IItem)item).EvaluatedIncludeEscaped;
         }
 
+        public List<ProvenanceResult> GetItemProvenance(string itemValue)
```

Comment:
Doc comments for all of these please. I may also want `remarks` sections explaining use cases.